cmake/SetupCodeChecks.cmake

```diff
@@ -654,7 +681,17 @@ macro(blt_add_uncrustify_target)
                             " for style check targets. "
                             " Current uncrustify executable: '${UNCRUSTIFY_EXECUTABLE}' "
                             " Current uncrustify version is: ${_uncrustify_version}."    )
-        endif()        
+        endif()
+
+        if(BLT_REQD_UNCRUSTIFY_VER)
+            # The user may only specify a part of the version (e.g. just the maj ver)
+            # so check for substring
+            string(FIND ${_uncrustify_version} ${BLT_REQD_UNCRUSTIFY_VER} VERSION_MATCHES)
+            if (VERSION_MATCHES EQUAL -1)
+                message(FATAL_ERROR "blt_add_uncrustify_target: uncrustify ${BLT_REQD_UNCRUSTIFY_VER} is required, found ${_uncrustify_version}")
+            endif()
+        endif()
+    endif() 
```

Comment:
This diff looks like you have added an unmatched endif? If not, can you fix the indenting for it.